src/main/java/careeratlas/backend/Service/JobSearchServiceImpl.java

```diff
@@ -1,34 +1,39 @@
 package careeratlas.backend.Service;
 
-import careeratlas.backend.Domain.JobResponse;
+import careeratlas.backend.Domain.IndeedResponse;
 import careeratlas.backend.Domain.JobSearch;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
 import org.springframework.stereotype.Service;
 
-import java.util.ArrayList;
+import java.io.IOException;
 import javax.ws.rs.client.*;
 import javax.ws.rs.core.MediaType;
 import javax.ws.rs.core.Response;
-import java.util.HashMap;
 
 @Service
 public class JobSearchServiceImpl implements JobSearchService {
 
     @Override
-    public ArrayList<JobResponse> searchJobs(String jobTitle, String jobType, String distance, String location){
+    public IndeedResponse searchJobs(String jobTitle, String jobType, String distance, String location){
         JobSearch jobSearch = new JobSearch(location, jobTitle, distance, jobType);
```

Comment:
when you say move it up a level do you mean create it on line 15 and then add everything inside the searchJob function?

Reply:
I mean whoever calls this method should be in charge of building the `JobSearch` object. Level I meant in the stack. You should always strive to deal with an object rather than fields as early as possible. Obviously if a class isnt dealing with "search" logic that wouldnt build a `JobSearch` object. But as soon as it makes sense its generally better to deal with objects.